### ISSUE TITLE
Ability to just change the brightness of the lights

### ISSUE THREAD

**Issue by Dustinhoefer** [ZeroOne3010/yetanotherhueapi#3]
Currently there is no support to only change the brightness of the lights. Could this be implemented?

Thanks!

**Comment by Dustinhoefer** [comment]
I just checked out the project and made a setter for the brightness in the state class. Any real reason why the variables are final? Can't we just make them mutable and create a setter?

**Comment by ZeroOne3010** [comment]
Hey, thanks for opening this issue! :smiley: The choice between immutable and mutable objects is mostly a personal preference, and my preference tends towards using immutable objects when possible. There are trade-offs in both cases. But let me see what I can come up with concerning the brightness setting...

**Comment by Dustinhoefer** [comment]
Thank you very much!
Will check it out later :) 
Great project 👍

**Comment by ZeroOne3010** [comment]
Great, thanks @Dustinhoefer! :) I added a `setBrightness(int)` method into the `Light` interface. It mimics the existing `turnOn()` and `turnOff()` methods. I realized that all Hue lights allow the brightness to be changed, so that was a natural place. Not all of them support for example color, so color related operations would be better kept in the `State` class. By modifying the `Light` interface I also feel like I'm respecting the philosophy that I wrote down in the README file: that essential functions should be done "in an intuitive and usable way for the programmer". :relaxed:
